r/clean.R

```diff
@@ -645,19 +645,45 @@ df <- df |> dplyr::mutate(
     .default = as.character(thyroid_surgery)
   )
 )
-df$thyroid_surgery |> table()
-
-
-## thyroid_surgery_lymph_node_pathology
-
-
-
-
 ## Free text fields
 ##
 ## presentation
 ## routine_review_management_revised_reason_other
 ## thyroid_surgery_lymph_node_pathology_other_type
-
 ## Finally save the data
 saveRDS(df, file = paste(r_dir, "clean.rds", sep = "/"))
+transform(df,
+  referral_source = factor(replace(as.character(referral_source),
+    list = !referral_source %in% c("GP"),
+    values = "secondary care"
+  ))
+)
+## converting other sources of referral to secondary care
+df$referral_source[df$referral_source %in% c("A&E", "Acute Medicine", "Cardiology", "Endocrinology", "Gastroenterology", "Gynaecology", "Heamatology", "Inpatient", "Oncology", "Oral & Maxillofacial", "Other", "Respiratory", "Rheumatology", "Urology")] <- "secondary care"
+table(df$referral_source)
+df$nodule_ultrasound_u_stage[df$nodule_ultrasound_u_stage %in% c("Not reported")] <- "NA"
+df$nodule_fna_thy[df$nodule_fna_thy %in% c("Not Applicable")] <- "NA"
```

Comment:
This is one way of recoding the data but you have three lines to do so. This line just replaces the string `Not Applicable` with another string `NA` and so its not actually what R recognises as missing. To get it as a "missing string" you actually need to use...

```
df$nodule_fna_thy[df$nodule_fna_thy %in% c("Not Applicable")] <- NA_character_
```

However, in the code further up we already have some similar tidying up of  `clinical_assessment` which uses the `dplyr::mutate()` and `dplyr::case_when()` functions.

And we can replace any instances of `Not Applicable` with `NA_character_`. The final step is to work out what we want values that _aren't_ `Not Applicable` to be and looking at your next two lines of code the common thing I can see is that you are replacing `Thy1` and `Thy1c` with `Thy1` and `Thy2` and `Thy2c` with `Thy2`.

Now you could write your `dplyr::case_when()` as...


```R
nodule_fna_thy = dplyr::case_when(nodule_fna_thy == "Not Applicable" ~ NA_character_,
    "Thy1" ~ "Thy1",
    "Thy1c" ~ "Thy1",
    "Thy2" ~ "Thy2",
    "Thy2c" ~ "Thy2")
```

And whatever the value on the left of `~` gets replaced with whatever is on the right.

But that is a lot of typing and in this instance there is a trick we can use because there is a common theme to the replacement you want to do which is that you are in essence taking the first four characters of whatever value `module_fna_thy` is and dropping the `c` from the end.

So we can use the special `.default` argument to say what to do if the value of `module_fna_thy` _isn't_ `Not Applicable` and if that is the case we take the first four characters, and we do that using [`stringr::str_sub()`](https://stringr.tidyverse.org/reference/str_sub.html) which takes a string, in this case the column `nodule_fna_thy` and returns the sub-string (`str_sub`) from the first position `1` to the fourth `4`.

```R
nodule_fna_thy = dplyr::case_when(nodule_fna_thy == "Not Applicable" ~ NA_character_,
    .default = stringr::str_sub(nodule_fna_thy, 1, 4)
)
```

Now the last step is to make sure this sits within the `dplyr::mutate()` call that starts on line 445, you need to put a `,` after the previous `clinical_assessment` tidying (where the `)` closes for `case_when(` on line 459 and then insert the above.